dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/deprecated/tracker/TrackedEntityInstanceCriteriaMapper.java

```diff
@@ -202,6 +184,45 @@ public TrackedEntityQueryParams map(TrackedEntityInstanceCriteria criteria) {
     return params;
   }
 
+  private Set<OrganisationUnit> validateOrgUnits(
+      TrackedEntityInstanceCriteria criteria,
+      Set<OrganisationUnit> possibleSearchOrgUnits,
+      User user) {
+
+    Set<OrganisationUnit> organisationUnits = new HashSet<>();
+
+    if (!criteria.getOrgUnits().isEmpty()
+        && (criteria.getOuMode() == ACCESSIBLE || criteria.getOuMode() == CAPTURE)) {
+      throw new IllegalQueryException(
+          String.format(
+              "ouMode %s cannot be used with orgUnits. Please remove the ou parameter and try again.",
+              criteria.getOuMode()));
+    }
+
+    for (String orgUnit : criteria.getOrgUnits()) {
+      OrganisationUnit organisationUnit = organisationUnitService.getOrganisationUnit(orgUnit);
+
+      if (organisationUnit == null) {
+        throw new IllegalQueryException("Organisation unit does not exist: " + orgUnit);
+      }
+
+      if (user == null) {
```

Comment:
I was doubting a bit in this one, since in theory the `currentUserService` could return null in these two places:
 - https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserStore.java#L424
 - https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/CurrentUserUtil.java#L49

Reply:
So my question is more:
why here we have this check and here EventRequestToSearchParamsMapper we don't?
And
Does the error message make sense? Can a login solve the issue or if this happen there is a bug somewhere?

Maybe on this last one @netroms would know better

Reply:
@netroms I removed the check to be able to merge the PR, but the question still stands.
Do we need to handle the case of a potential null user when calling `currentUserService.getCurrentUser()`? If so, should we advise the user to log in again? Will the issue be resolved after the re-login?